cvat-canvas/src/typescript/interactionHandler.ts

```diff
@@ -207,11 +214,16 @@ export class InteractionHandlerImpl implements InteractionHandler {
     private initInteraction(): void {
         if (this.interactionData.crosshair) {
             this.addCrosshair();
+        } else if (this.crosshair) {
+            this.removeCrosshair();
         }
-
         if (this.interactionData.enableThreshold) {
             this.addThreshold();
+        } else if (this.threshold) {
+            this.threshold.remove();
+            this.threshold = null;
         }
+        document.getElementById('cvat_canvas_wrapper').style.cursor = 'crosshair';
```

Comment:
Where do you reset it?

Reply:
To enable tools locking, im passing `interactionData` without `intermediateShape` param just to call `initInteraction` method of `interactionHandler`. As i see in `canvasView.ts` when interactionData has no `intermediateShape` is disables crosshair cursor. To keep it, im using  `document.getElementById('cvat_canvas_wrapper').style.cursor = 'crosshair';`. So reset occurs in `canvasView.ts` when tool is deactivated.

Reply:
I've removed this line and updated logic in canvasView.ts by checking onChangeBlock field.